src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestUnreachable.ll

```diff
@@ -0,0 +1,42 @@
+﻿define i64 @Microsoft__Quantum__Testing__QIR__TestUnreachable__body(i64 %a, i64 %b) {
+entry:
+  %c = add i64 %a, %b
+  %0 = icmp eq i64 %c, 5
+  br i1 %0, label %then0__1, label %else__1
+
+then0__1:                                         ; preds = %entry
+  ret i64 %c
+
+then0__1__unreachable__1:                         ; No predecessors!
+  %d = add i64 2, %a
+  %1 = call %String* @__quantum__rt__string_create(i8* getelementptr inbounds ([14 x i8], [14 x i8]* @0, i32 0, i32 0))
+  call void @__quantum__rt__fail(%String* %1)
+  unreachable
+
+then0__1__unreachable__2:                         ; No predecessors!
+  call void @__quantum__rt__string_update_reference_count(%String* %1, i32 -1)
```

Comment:
Note that the ref decrement calls in these unreachable blocks are generated because each of these blocks exit from the scope of the original allocation, either via a terminal statement or control reaching the end of the block.

This behavior is not new in this PR. The only difference is now we explicitly label these unreachable blocks to ensure that we end up with valid LLVM IR.